src/libs/actions/IOU.js

```diff
@@ -41,115 +47,131 @@ function getIOUReportsForNewTransaction(requestParams) {
                         hasOutstandingIOU: true,
                     };
 
-                    // Second, the IOU report needs updated with the new IOU details too
+                    // Second, the IOU report needs to be updated with the new IOU details too
                     const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${reportData.reportID}`;
                     iouReportsToUpdate[iouReportKey] = Report.getSimplifiedIOUReport(reportData, chatReportID);
                 }
             });
 
-            // Now, merge the updated objects into our store
             Onyx.mergeCollection(ONYXKEYS.COLLECTION.REPORT, chatReportsToUpdate);
-            return Onyx.mergeCollection(ONYXKEYS.COLLECTION.REPORT_IOUS, iouReportsToUpdate);
+            Onyx.mergeCollection(ONYXKEYS.COLLECTION.REPORT_IOUS, iouReportsToUpdate);
         })
-        .catch(() => Onyx.merge(ONYXKEYS.IOU, {error: true}))
```

Comment:
Are we removing `.catch()`-es because the jsonCodes from the API are really the only thing we have to go on? Nothing's being thrown in the JS layer that could be caught?

Reply:
Can of worms, but generally speaking there's only a couple reasons why you end up in a catch and they are:

**`fetch()` errors**

![2022-03-22_09-29-40](https://user-images.githubusercontent.com/32969087/159561275-dfddcfd2-8428-4be6-a40b-ad9e9b361428.png)

**force promise to be rejected by throwing an exception because you are doing something bad in a `.then()` or intentionally threw a `new Error()`**

![2022-03-22_09-33-12](https://user-images.githubusercontent.com/32969087/159561267-3d44d45a-681b-4826-ad9a-6f4f0f5c9ce3.png)

For the fetch errors we should be handling them globally in the network layer here for persisted things like report comments

https://github.com/Expensify/App/blob/2969ed7bd0635a98d07520a1a3ba09c985fdb7c3/src/libs/Network.js#L71-L76

Any persisted `fetch()` requests should not ever be getting rejected they are just retried or dumped at some point.

For the ones that _aren't persisted_  (one's being modified in this PR) we could trigger the `onError()` callback here where the request will get retried for a while and then eventually we **do** throw an error

https://github.com/Expensify/App/blob/2969ed7bd0635a98d07520a1a3ba09c985fdb7c3/src/libs/Network.js#L236-L259

Which lands us here and rejects the promise (which would lead to an unhandled promise rejection unless we add a catch when we call the method)

https://github.com/Expensify/App/blob/2969ed7bd0635a98d07520a1a3ba09c985fdb7c3/src/libs/API.js#L193-L207

So a promise _could_ be rejected in theory. But **if we are offline we should not be making the request in the first place**. 

Having fun yet?

Technically, we could probably argue that in the current state of the app all API methods should be using a `catch()` because there's always a chance that you were online -> make a request -> go offline -> get a fetch error. But barring any accidental exceptions caused by bad code _the only error we should expect to see_ is the `CONST.ERROR.API_OFFLINE` error. 

If you look at the catches I want to remove here. We can no longer make that assumption because we added other reasons why these promises could be rejected (which I'm undoing) or we explicitly throw errors in the `.then()` which isn't really necessary and makes the reason behind the `.catch()` ambiguous. Was it a rare "offline" error or the one we explicitly throw? Do we bother to check?

Basically, this whole business is way more complicated than it has to be and the complexity has led to a lot of non standard error handling.